### PR TITLE
[STAKAN-256] error value (NaN) showing up when deleting a number in Staking manager

### PR DESCRIPTION
**Pull request by fox-on-rocket**
Linked with https://github.com/Ankr-network/stkr-web3/pull/359 and https://github.com/Ankr-network/stkr-web3/pull/360

### CODE REVIEW

**Review by Antroll — APPROVED** [review]

File: src/modules/stake/components/StakeForm/StakeForm.tsx [hunk 1]
```diff
@@ -38,6 +42,9 @@ export interface IStakeFormComponentProps {
   stakeInfo?: string;
 }
 
+const getAmountNum = (amount?: ReactText): number =>
+  amount ? Math.abs(+amount) : 0;
```
Comment (BigBadAlien) [inline src/modules/stake/components/StakeForm/StakeForm.tsx:46]: Remove Math.abs
Add validation "Value can’t be less than 0"
If the amount is not valid then earnings is zero

File: src/common/utils/isNumGTEZero.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export function isNumGTEZero(val?: number | unknown): boolean {
```
[the remark this answers is not in the document]
Comment (fox-on-rocket) [reply src/common/utils/isNumGTEZero.ts]: Outdated

**Review by BigBadAlien — APPROVED** [review]

File: src/modules/stake-polygon/screens/StakePolygon/StakePolygon.tsx [hunk 3]
```diff
@@ -60,7 +64,7 @@ export const StakePolygon = () => {
               </StakeDescriptionName>
 
               <StakeDescriptionValue>
-                {t('unit.~polygon', {
+                {t(isZeroAmount ? 'unit.matic-value' : 'unit.~polygon', {
```
Comment (BigBadAlien) [inline src/modules/stake-polygon/screens/StakePolygon/StakePolygon.tsx:67]: FYI. You can use "{value, plural, =0 {# Matic} other {~# Matic}}",

Comment (BigBadAlien) [reply src/modules/stake-polygon/screens/StakePolygon/StakePolygon.tsx:67]: It's a way and not required

**Pull request merged**